development

Milerius opened the pull request.

Milerius committed:
feat(http): add restinio dependencies

Milerius committed:
feat(http): starting http server

Milerius committed:
feat(http): upgrade restinio target

Milerius committed:
feat(http): silence asio warnings

Milerius committed:
feat(http): upgrade restinio target

Milerius committed:
feat(http): add http_port to config

Milerius committed:
feat(http): add basic unit tests for the server skeleton

Milerius committed:
feat(http): split unit tests in source file

Milerius committed:
improvements(http): remove unwanted include

Milerius committed:
feat(price-service): add all price function

Milerius committed:
improvements(price-service): st_price is now an std::uint64_t

Milerius committed:
improvements(price-service): check for ETH base symbol

Milerius committed:
improvements(price-service): fixed narrowing conversion

Milerius committed:
improvements(price-service): fixed narrowing conversion

Milerius committed:
improvements(price-service): fixed narrowing conversion

Milerius committed:
feat(price-service): add jl777 coins repos in dependencies

Milerius committed:
feat(price-service): copy coins config into binary dir

Milerius committed:
fix(cmake): rename targets for not conflicting with cpptask

Milerius committed:
improvements(price-service): use jl777 coins configuration for check ercs

DCPRevere commented on src/price/service.price.platform.cpp:
```diff
@@ -60,15 +61,38 @@ namespace antara::mmbot
+            if (mmbot_config_.base_ercs_registry.at(current_pair.base.symbol.value())) {
+                current_price_str = fmt::format("{:.18f}", static_cast<double>(current_price.value()) / g_factor);
+            } else {
+                current_price_str = fmt::format("{:.8f}", static_cast<double>(current_price.value()) / g_factor);
```
Can we put this logic somewhere we can reuse it? As a method of the asset struct?

Milerius replied:
Yes good idea

DCPRevere replied:
I realise that it's complicated by the `base_ercs_registry`, but maybe we can put this information inside the asset struct.

Milerius replied:
the information about ercs is inside a configuration file: coins.json written by @jl777

That's why it's parsed once at the beginning of the program and can be used from everywhere

but maybe the name is bad and should be renamed

Milerius committed:
improvements(price-service): split get_price_as_decimal in an another function

Milerius committed:
improvements(price-service): add rounding

Milerius committed:
improvements(price-service): add unit tests for antara utils

Milerius committed:
feat(http): re enable tests

Milerius committed:
feat(price): add helpers

Milerius committed:
feat(price): managing price without floating point

Milerius committed:
style: adjust coding style

Milerius committed:
improvements: upgrade restinio 0.5.1 to 0.5.1.1

Milerius committed:
feat(app): start skeleton for the app

Milerius committed:
style: remove unwanted comments

Milerius committed:
tests: using fixture for http unit tests (dry)

Milerius committed:
tests: add some BDD tests

Milerius committed:
tests: finishing bdd tests for service

Milerius committed:
feat(price): remove last floating point utilisation for price

Milerius committed:
feat(mmbot): add version dump at the start of the app and more log

Milerius committed:
feat(mmbot): add more important changement

- parsing float api inpoint into string
- start http rest call getprice
- add more unit tests

Milerius committed:
style: rename base_ercs_registry to precision_registry

Milerius committed:
tests: add more unit tests for formatting

Milerius committed:
feat(price-format): add abseil and using uint128_t now

Milerius commented:
@DCPRevere I Commented some part of your code that you need to update on this pull request please see: https://github.com/KomodoPlatform/antara-makerbot/pull/15/commits/b028dc96c3c934d75e40395573e1f7a34e92282d

Milerius committed:
feat(price-format): update operator== due to abseil

Milerius committed:
feat(devops): update script build

Milerius committed:
feat(devops): update script build

Milerius committed:
feat(price format): add more tests

Milerius committed:
feat(price format): get raw string from the json

Milerius committed:
feat(price format): using bcmath for commercial rounding

Milerius committed:
fix(compilation): fix building linux

Milerius committed:
style: remove -Weffc++

Milerius committed:
fix(windows compilation): add nominmax definition
